src/com/facebook/buck/cxx/toolchain/WindowsPreprocessor.java

```diff
@@ -61,22 +59,13 @@ private static String prependIncludeFlag(String includeRoot) {
 
   @Override
   public Iterable<String> prefixHeaderArgs(Path prefixHeader) {
-    throw new UnsupportedOperationException("prefix header not supported by " + getClass());
-    // TODO(steveo) Should be easy to add support; will try @ later time,
-    // when I can test w/ Windows.
-    // "Forced Include": https://msdn.microsoft.com/en-us/library/8c5ztk84.aspx
-    // Space is allowed between flag and its pathname argument.
-    // E.g. something like this (space allowed between flag and its argument):
-    // return ImmutableList.of("/FI", resolver.getAbsolutePath(prefixHeader).toString());
+    return ImmutableList.of("/Yc", prefixHeader.toString());
   }
 
   @Override
   public Iterable<String> precompiledHeaderArgs(Path pchOutputPath) {
-    throw new UnsupportedOperationException("precompiled header not supported by " + getClass());
-    // TODO(steveo) Should be easy to add support; will try @ later time,
-    // when I can test w/ Windows.
-    // https://msdn.microsoft.com/en-us/library/z0atkd6c.aspx
-    // E.g. something like this flag (no space between "/Yu" and its argument):
-    // return ImmutableList.of("/Yu" + pchOutputPath);
+    String hFilename = pchOutputPath.getFileName().toString();
+    String pchFilename = hFilename.substring(0, hFilename.length() - 4) + ".pch";
```

Comment:
Are you trying to strip the file extension here?
Perhaps we can use https://google.github.io/guava/releases/19.0/api/docs/com/google/common/io/Files.html#getNameWithoutExtension(java.lang.String)